sentinel5dl/__main__.py

```diff
@@ -7,30 +7,149 @@
 :url: https://emissions-api.org
 :license: MIT
 '''
-
+import argparse
+import iso8601
 import logging
+import textwrap
 
 import sentinel5dl
 from sentinel5dl import search, download
 
+PRODUCTS = (
+    'L1B_IR_SIR',
+    'L1B_IR_UVN',
+    'L1B_RA_BD1',
+    'L1B_RA_BD2',
+    'L1B_RA_BD3',
+    'L1B_RA_BD4',
+    'L1B_RA_BD5',
+    'L1B_RA_BD6',
+    'L1B_RA_BD7',
+    'L1B_RA_BD8',
+    'L2__AER_AI',
+    'L2__AER_LH',
+    'L2__CH4___',
+    'L2__CLOUD_',
+    'L2__CO___',
+    'L2__HCHO__',
+    'L2__NO2__',
+    'L2__NP_BD3',
+    'L2__NP_BD6',
+    'L2__NP_BD7',
+    'L2__O3_TCL',
+    'L2__O3___',
+    'L2__SO2__',
+)
+
+PRODUCTS_STR = textwrap.fill(', '.join(PRODUCTS),
+                             subsequent_indent='  ',
+                             initial_indent='  ')
+
+PROCESSING_LEVELS = (
+    'L1B',
+    'L2'
+)
+
+
+def is_polygon(polygon):
+    '''Validate if the supplied polygon string is in the necessary format to be
+    used as part of a WKT polygon string.
+
+    :param polygon: Polygon string
+    :return: WKT polygon
+    '''
+    values = [value.strip() for value in polygon.split(',')]
+
+    # Polygon must be at least a triangle
+    if len(values) < 4:
+        raise ValueError()
+
+    # Check if we got float pairs
+    for value in values:
+        if len([float(x) for x in value.split()]) != 2:
+            raise ValueError('Polygon values must be pairs of numbers')
+
+    # Check if we got a closed polygon
+    if values[0] != values[-1]:
+        raise ValueError()
+
+    return f'POLYGON(({polygon}))'
+
+
+def validate_date_string(date_string):
+    '''Validate that the supplied argument is a valid iso8601 date-time string.
+
+    :param date_string: Date string to validate
+    :return: Supplied date string
+    '''
+    try:
+        iso8601.parse_date(date_string)
```

Comment:
`dateutils` is very cool when it comes to parsing arbitrary date strings. But here we want to allow only ISO8601. `dateutils.parse` would let a lot of invalid date strings pass. Example:

```
% python3 -m sentinel5dl --polygon '30 10, 40 40, 20 40, 10 20, 30 10' --begin_ts 2019-09-01T00:00:00.000Z \
    --end_ts 'Mon 04 Nov 2019 09:00:07 PM CET'
INFO:sentinel5dl:Searching for Sentinel-5 products
Traceback (most recent call last):
    …
    curl.perform()
pycurl.error: (22, 'The requested URL returned error: 400 Bad Request')
```